src/Kontent.Ai.ModelGenerator/ArgHelpers.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Kontent.Ai.Delivery.Abstractions;
+using Kontent.Ai.Management.Configuration;
+using Kontent.Ai.ModelGenerator.Core.Configuration;
+
+namespace Kontent.Ai.ModelGenerator;
+
+internal static class ArgHelpers
+{
+    private static readonly ProgramOptionsData ManagementProgramOptionsData = new ProgramOptionsData(typeof(ManagementOptions), "management-sdk-net");
+    private static readonly ProgramOptionsData DeliveryProgramOptionsData = new ProgramOptionsData(typeof(DeliveryOptions), "delivery-sdk-net");
+
+    private static readonly IDictionary<string, string> GeneralMappings = new Dictionary<string, string>
+    {
+        { "-n", nameof(CodeGeneratorOptions.Namespace) },
+        { "-o", nameof(CodeGeneratorOptions.OutputDir) },
+        { "-f", nameof(CodeGeneratorOptions.FileNameSuffix) },
+        { "-g", nameof(CodeGeneratorOptions.GeneratePartials) },
+        { "-b", nameof(CodeGeneratorOptions.BaseClass) }
+    };
+    private static readonly IDictionary<string, string> DeliveryMappings = new Dictionary<string, string>
+    {
+        { "-p", $"{DeliveryProgramOptionsData.OptionsName}:{nameof(DeliveryOptions.ProjectId)}" },
+        {"--projectid", $"{DeliveryProgramOptionsData.OptionsName}:{nameof(DeliveryOptions.ProjectId)}" }, // Backwards compatibility
+        { "-s", nameof(CodeGeneratorOptions.StructuredModel) },
+        { "-t", nameof(CodeGeneratorOptions.WithTypeProvider) }
+    };
+    private static readonly IDictionary<string, string> ManagementMappings = new Dictionary<string, string>
+    {
+        { "-p", $"{ManagementProgramOptionsData.OptionsName}:{nameof(ManagementOptions.ProjectId)}" },
+        {"--projectid", $"{ManagementProgramOptionsData.OptionsName}:{nameof(ManagementOptions.ProjectId)}" }, // Backwards compatibility
+        { "-m", nameof(CodeGeneratorOptions.ManagementApi) },
+        { "-k", $"{ManagementProgramOptionsData.OptionsName}:{nameof(ManagementOptions.ApiKey)}" },
+        { "--apikey", $"{ManagementProgramOptionsData.OptionsName}:{nameof(ManagementOptions.ApiKey)}" } // Backwards compatibility
+    };
+
+    private static readonly IEnumerable<string> AllMappingsKeys = GeneralMappings.Keys.Union(DeliveryMappings.Keys).Union(ManagementMappings.Keys);
+
+    public static IDictionary<string, string> GetSwitchMappings(string[] args)
+    {
+        return GeneralMappings
+            .Union(ContainsManageApiArg() ? ManagementMappings : DeliveryMappings)
+            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+
+        bool ContainsManageApiArg() =>
+            args.Where((value, index) => (value is "-m" or "--managementapi") && index + 1 < args.Length && args[index + 1] == "true").Any();
+    }
+
+    public static bool ContainsUnsupportedArg(string[] args)
```

Comment:
I am not a fan of inverted conditions/method names. 

`ContainValidArgs` might be better  - just and idea, not requirement.

Reply:
https://github.com/kontent-ai/model-generator-net/blob/feature/154_improve_user_experience/src/Kontent.Ai.ModelGenerator/ArgHelpers.cs#L52